release/models/system/openconfig-connections.yang

```diff
@@ -0,0 +1,291 @@
+          }
+          enum UDP6 {
+            description
+              "Index value indicating UDP over IPv6";
+          }
+          enum UDPL {
+            description
+              "Index value indicating UDP-Lite";
+          }
+          enum RAW {
+            description
+              "Index value indicating a RAW socket";
+          }
+        }
+        type string;
+      }
+      description
+        "The protocol of the connection.";
+    }
+
+    leaf local-address {
+      type oc-inet:host;
+      description
+        "The local address of the connection";
+    }
+
+    leaf local-port {
+      type oc-inet:port-number;
+      description
+        "The local port of the connection";
+    }
+
+    leaf remote-address {
+      type oc-inet:host;
+      description
+        "The address at the remote end of the connection";
+    }
+
+    leaf remote-port {
+      type oc-inet:port-number;
+      description
+        "The port at the remote end of the connection";
+    }
+
+    leaf tcp-state {
+      when "../protocol = 'TCP' or ../protocol = 'TCP6'";
+      type enumeration {
+        enum ESTABLISHED {
+          description
+            "The connection is established.";
+        }
+        enum SYN_SENT {
+          description
+            "A connection request has been sent.";
+        }
+        enum SYN_RECV {
+          description
+            "A connection request has been received.";
+        }
+        enum FIN_WAIT1 {
+          description
+            "The socket is closed, and the connection is shutting down.";
+        }
+        enum FIN_WAIT2 {
+          description
+            "The connection is closed, and the socket is waiting for a shutdown from the remote end.";
+        }
+        enum TIME_WAIT {
+          description
+            "The socket is waiting after closing to handle packets still in the network.";
+        }
+        enum CLOSE {
+          description
+            "The socket is not being used.";
+        }
+        enum CLOSE_WAIT {
+          description
+            "The remote end has shut down, and the socket is waiting for the local close.";
+        }
+        enum LAST_ACK {
+          description
+            "The remote end has shut down, and the socket is waiting for an acknowledgment of the last packet to be sent.";
+        }
+        enum LISTEN {
+          description
+            "The socket is listening for incoming connections.";
+        }
+        enum CLOSING {
+          description
+            "Both sockets are shut down but we still don't have all our data sent.";
+        }
+        enum UNKNOWN {
+          description
+            "The TCP state is unknown.";
+        }
+      }
+      description
+        "The current state of the TCP connection.";
+    }
+
+    leaf process-name {
+      type string;
+      description
+        "The name of the process associated with the connection.";
+    }
+
+    leaf pid {
+      type uint64;
```

Comment:
While `openconfig-procmon` defines a pid as `uint64`, I think this is too generous.  Linux/BSD as well as VxWorks based systems use `uint32` for PIDs/tasks often w/ a much lower max/default

```
$ cat /proc/sys/kernel/pid_max 
32768
```